marconi-sidechain/src/Marconi/Sidechain/CLI.hs

```diff
@@ -1,17 +1,25 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# OPTIONS_GHC -Wno-missing-import-lists #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
 module Marconi.Sidechain.CLI (
   CliArgs (..),
   parseCli,
+  parseCliArgs,
   Cli.getVersion,
   programParser,
 ) where
 
 import Cardano.Api qualified as C
+import Data.Aeson (FromJSON, ToJSON)
 import Data.List.NonEmpty (NonEmpty)
+import GHC.Generics (Generic)
 import Marconi.ChainIndex.CLI qualified as Cli
-import Marconi.ChainIndex.Node.Client.Retry (RetryConfig)
+import Marconi.ChainIndex.Node.Client.Retry (RetryConfig (..))
 import Marconi.ChainIndex.Types (
-  IndexingDepth,
-  ShouldFailIfResync,
+  IndexingDepth (..),
+  ShouldFailIfResync (..),
```

Comment:
These (..) aren't needed I think. Weird that you didn't get any warning for those as we're using `Wmissing-import-lists`.

Reply:
The constructors are being used, but in a non-obvious way: for deriving `Generic`

Reply:
If I move the derivations as Nicolas suggested, these will go away.